src/pkg/packaging-tools/sharedFramework/sharedFramework.csproj

```diff
@@ -0,0 +1,153 @@
+  -->
+  <Target Name="RemovePlatformManifests"
+          BeforeTargets="
+            _HandlePublishFileConflicts;
+            _HandlePackageFileConflicts">
+    <ItemGroup>
+      <PackageConflictPlatformManifests Remove="@(PackageConflictPlatformManifests)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="GenerateNetCoreAppRuntimeConfig"
+          Condition="'$(GenerateNetCoreAppRuntimeConfig)' == 'true'"
+          BeforeTargets="Publish">
+    <MakeDir Directories="$(PublishDir)" />
+    
+    <PropertyGroup>
+      <_runtimeConfigContent>
+      <![CDATA[
+{
+  "runtimeOptions": {
+    "tfm": "$(TargetFramework)",
+    "framework": {
+      "name": "Microsoft.NETCore.App",
+      "version": "$(SharedFrameworkNugetVersion)"
+    }
+  }
+}
+      ]]>
+      </_runtimeConfigContent>
+    </PropertyGroup>
+    
+    <WriteLinesToFile
+      File="$(PublishDir)$(SharedFrameworkName).runtimeconfig.json"
+      Lines="$(_runtimeConfigContent)"
+      Overwrite="true" />
+  </Target>
+
+  <Target Name="GenerateVersionFile"
+          DependsOnTargets="GetPackageReference"
+          BeforeTargets="Publish">
+    <!-- Generate .version file -->
+    <ItemGroup>
+      <VersionLines Include="$(LatestCommit)" />
+      <VersionLines Include="%(PackageReference.Version)" />
+    </ItemGroup>
+
+    <WriteLinesToFile
+      File="$(PublishDir).version"
+      Lines="@(VersionLines)"
+      Overwrite="true" />
+  </Target>
+
+  <!-- Clean up artifacts that dotnet-publish generates which we don't need -->
+  <Target Name="DeleteUnnecessaryPublishOutputs" AfterTargets="Publish">
+    <ItemGroup>
+      <ToDelete Include="$(PublishDir)apphost$(ExeSuffix)" />
+      <ToDelete Include="$(PublishDir)$(LibPrefix)comhost$(LibSuffix)" />
+      <ToDelete Include="$(PublishDir)$(LibPrefix)hostfxr$(LibSuffix)" />
+      <ToDelete Include="$(PublishDir)$(LibPrefix)ijwhost$(LibSuffix)" />
+    </ItemGroup>
+
+    <Delete Files="@(ToDelete)" />
+  </Target>
+
+  <Target Name="GenerateRuntimeGraph"
+          Condition="'$(GenerateRuntimeGraph)' == 'true'"
+          AfterTargets="Publish">
+    <PropertyGroup>
+      <SharedFrameworkAssetsFile>$(IntermediateOutputPath)/project.assets.json</SharedFrameworkAssetsFile>
+      <SharedFrameworkDepsFile>$(PublishDir)$(ProjectDepsFileName)</SharedFrameworkDepsFile>
+      <RuntimeGraphGeneratorRuntime Condition="'$(OSGroup)'=='Windows_NT'">win</RuntimeGraphGeneratorRuntime>
+      <RuntimeGraphGeneratorRuntime Condition="'$(OSGroup)'=='OSX'">osx</RuntimeGraphGeneratorRuntime>
+      <RuntimeGraphGeneratorRuntime Condition="'$(OSGroup)'=='FreeBSD'">freebsd</RuntimeGraphGeneratorRuntime>
+      <RuntimeGraphGeneratorRuntime Condition="'$(RuntimeGraphGeneratorRuntime)'==''">linux</RuntimeGraphGeneratorRuntime>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <TrimPkgsFromDeps Include="runtime.$(PackageTargetRid).microsoft.netcore.dotnetapphost" />
```

Comment:
Do we still have these pacakges in the deps?  I thought we flattened it.

Reply:
`Microsoft.WindowsDesktop.App` and its runtime package don't have these dependencies, so currently this doesn't do anything.

`Microsoft.NETCore.App` still has them... I also remember this coming up before but forget what happened. Maybe there was just no particular reason to trim them ahead of time and instead we'd just wait for the package to disappear. (https://github.com/dotnet/core-setup/issues/5663 is a step towards that.)

I'll remove the trimming from this PR. When this infra is used for netcoreapp, the solution will be to remove the dependencies from the package itself.